staging/src/k8s.io/apiserver/pkg/storage/storagebackend/config.go

```diff
@@ -28,7 +28,8 @@ const (
 	StorageTypeUnset = ""
 	StorageTypeETCD3 = "etcd3"
 
-	DefaultCompactInterval = 5 * time.Minute
+	DefaultCompactInterval      = 5 * time.Minute
+	DefaultDbMetricPollInterval = 30 * time.Second
```

Comment:
Could you please explain why is the polling interval set to 30s?  Just curious if there is a specific reason.

Reply:
How expensive is the call? If it's cheap, then have you considered using a GaugeFunc?

Reply:
@wenjiaswe I don't have a specific reason that leads to 30s. A normal k8s cluster has 3 to 5 etcd servers, so 5 calls every 30s should be negligible. On the other hand I don't feel we need to update this metric very often as its value changes rather slowly.

@logicalhan Using GaugeFunc means we make rpc call to each etcd server when user hits metrics endpoint? This is probably too much latency?

Reply:
Yes, db size change is usually very slow, that why I was thinking and wonder if we want a longer interval. But I am fine with what it is now as long as it does not impact performance. Thanks

Reply:
I still think a GaugeFunc is probably more idiomatic. You can cache the value and not call out to etcd if the value hasn't TTL'd yet (the TTL would be the same as the poll interval currently)

Reply:
@logicalhan I did not find an easy way to incorporate GaugeFunc with a gauge vector metric?

Reply:
Yeah, it turns out it isn't exposed.